docs/deployment/resource-providers/yarn.md

```diff
@@ -0,0 +1,200 @@
+
+### Application Mode
+
+Application Mode will launch a Flink cluster on YARN, where the main() method of the application jar gets executed on the JobManager in YARN.
+The cluster will shut down as soon as the application has finished. You can manually stop the cluster using `yarn application -kill <ApplicationId>` or by cancelling the Flink job.
+
+{% highlight bash %}
+./bin/flink run-application -t yarn-application ./examples/streaming/TopSpeedWindowing.jar
+{% endhighlight %}
+
+To unlock the full potential of the application mode, consider using it with the `yarn.provided.lib.dirs` configuration option
+and pre-upload your application jar to a location accessible by all nodes in your cluster. In this case, the 
+command could look like: 
+
+{% highlight bash %}
+./bin/flink run-application -t yarn-application \
+	-Dyarn.provided.lib.dirs="hdfs://myhdfs/my-remote-flink-dist-dir" \
+	hdfs://myhdfs/jars/my-application.jar
+{% endhighlight %}
+
+The above will allow the job submission to be extra lightweight as the needed Flink jars and the application jar
+are  going to be picked up by the specified remote locations rather than be shipped to the cluster by the 
+client.
+
+### Per-Job Cluster Mode
+
+The Per-job Cluster mode will launch a Flink cluster on YARN, then run the provided application jar locally and finally submit the JobGraph to the JobManager on YARN. If you pass the `--detached` argument, the client will stop once the submission is accepted.
+
+The YARN cluster will stop once the job has stopped.
+
+{% highlight bash %}
+./bin/flink run -m yarn-cluster --detached ./examples/streaming/TopSpeedWindowing.jar
+{% endhighlight %}
+
+With `bin/flink run -m yarn-cluster`, you can run the command line interface in YARN cluster mode, allowing to submit jobs to an ad-hoc YARN cluster. The command line options of the YARN session described below are then also available. They are prefixed with a `y` or `yarn` (for the long argument options).
+
+### Session Mode
+
+We describe deployment with the Session Mode in the [Getting Started](#getting-started) guide at the top of the page.
+
+The Session Mode has two operation modes:
+- **attached mode** (default): The yarn-session.sh client submits the Flink clsuter to YARN, but the client keeps running, tracking the state of the cluster. If the cluster fails, the client will show the error. If the client gets terminated, it will signal the cluster to shut down as well.
+- **detached mode**: The yarn-session.sh client submits the Flink cluster to YARN, then the client returns. Another invocation of the client, or YARN tools is needed to stop the Flink cluster.
+
+The session mode will create a hidden YARN properties file in `/tmp/.yarn-properties-<username>`, which will be picked up for cluster discovery by the command line interface when submitting a job.
```

Comment:
Maybe also mention how to explicitly specify a yarn cluster when submitting a job (e.g. if you have multiple Flink session cluster running).